packages/api/endpoints/granules.js

```diff
@@ -108,21 +109,38 @@ const update = async (req, res) => {
     knex = await getKnexClient(),
   } = req.testContext || {};
   const body = req.body || {};
+  const { retries = 0, status, ...restOfBody } = body;
 
   let existingGranule;
+
   try {
-    existingGranule = await granuleModel.get({ granuleId: body.granuleId });
+    existingGranule = await pRetry(
+      async () => await granuleModel.get({ granuleId: body.granuleId }),
+      {
+        retries,
+      }
+    );
   } catch (error) {
     if (error instanceof RecordDoesNotExist) {
       return res.boom.notFound(`No granule found to update for ${body.granuleId}`);
     }
     return res.boom.badRequest(errorify(error));
   }
 
+  let updatedBody = { status, ...body };
+  let message = '';
+
+  // Only set status to queued if granule was running
+  // Prevents race condition of setting granule to queued after completed
+  if (status === 'queued' && existingGranule.status !== 'running') {
```

Comment:
I'm concerned that this won't always work as you're expecting in duplicate/reingest scenarios as it's possible to have 

Execution 1: 
<some statuses> 
complete

Execution 2: 
Fails to write running status due to write redrive
<this code is called> and status isn't updated. 

There's complexity around how we're doing checks against executions in the sf write lambda and associated logic for this reason, and while I could see us offloading that to the users of the API (e.g. disco granules here), if we're doing the check I think we need to include some way of evaluating a reingest here.